akka-http-core/src/main/scala/akka/http/impl/settings/ConnectionPoolSettingsImpl.scala

```diff
@@ -33,6 +36,7 @@ object ConnectionPoolSettingsImpl extends SettingsCompanion[ConnectionPoolSettin
   def fromSubConfig(root: Config, c: Config) = {
     ConnectionPoolSettingsImpl(
       c getInt "max-connections",
+      c getIntOpt "min-connections" getOrElse 0,
```

Comment:
It will be backwards-compatible, yes.
So you can bump the Akka version and it'll work, this is because bumping the version will also bring in the "new" reference.conf which includes the new setting.